### DISCUSSION TITLE
storage upload "new row violates row-level security policy"

### DISCUSSION THREAD

**Discussion by mazaoyong** [supabase/supabase#30510] [Questions] [answered]
When uploading files, I encountered the error 'new row violates row-level security policy'. Below is a screenshot of my configuration.
![image](https://github.com/user-attachments/assets/e2170cca-d897-4a1a-b857-7c1aadd7130e)
My select, insert, update and delete strategies are all the same:
![image](https://github.com/user-attachments/assets/afaf8d31-5e5e-4165-865c-a3fa3160909d)
Here is my code:
```javascript
"use client";
export const UploadButton = () => {
  return (
    <FileButton
      onChange={async (file) => {
        const { data } = await spClient.browser.auth.getSession();
        console.log(333, data);
        if (data && file) {
          spClient.js.storage.from("test").upload(`${data?.session?.user?.id}/${Date.now()}_${file?.name}`,file);
        }
      }}
      accept="image/png,image/jpeg"
    >
    </FileButton>
  );
};
```

**Comment by GaryAustin1** [comment, marked as answer]
You don't show your insert policy (which would need a WITH CHECK).
You also need to make sure you getSession data has a user jwt in the access_token.

**Reply by mazaoyong** [reply]
My select, insert, update and delete strategies are all the same.
And the data?.session?.user data exists.

**Reply by GaryAustin1** [reply]
Are the same or your insert uses WITH CHECK instead of USING?
Also having data.session.user does not mean you have a user session with access_token and user_token.  You need the access_token jwt to be there for auth.uid() type functions to work.

**Reply by mazaoyong** [reply]
Thanks, I found the reason. In my Next.js project, the middleware wasn't working due to incorrect file placement, which resulted in the auth cookie not being stored successfully.
